src/ex.rs

```diff
@@ -457,9 +459,13 @@ fn capture_lockfile(
     path: &Path,
     toolchain: &Toolchain,
 ) -> Result<()> {
-    let args = &["generate-lockfile", "--manifest-path", "Cargo.toml"];
+    let args = &["generate-lockfile", "--manifest-path", "Cargo.toml", "-Z", "offline"];
```

Comment:
I expect we'll want to store whether -Zoffline works somewhere -- probably in `toolchain`; we could make it "probably work" for CI and `starts_with("nightly")` for `Toolchain::Dist`. That way we avoid the somewhat expensive failure here (docker container creation, primarily).

Reply:
That makes sense, I assumed run_cargo was a low cost command. If it is not, than how best to test if cargo has `-Z offline` and where to sotor that knowledge?

Reply:
Well, "low cost" to an extent -- it creates a docker container. So the ~500 milliseconds or so (no idea, just assumption) that this will waste is 1.4 hours of runtime over the 10,000 crates that we have (worst case).

I think the best thing to do is to use a heuristic as I said above; CI will probably work, nightly should, and beta/stable will not. I'd add a function to `Toolchain` that returns a bool (something like `fn use_unstable_features(&self) -> bool`).

Does that help?